_posts/2024-04-10-benchmarks-over-time.md

```diff
@@ -0,0 +1,548 @@
+This approach avoids many comparison pitfalls, and also provides several valuable data points to consider when selecting a system.
+
+#### How fast is it improving? 
+
+When you select a data processing system, it is important to think about the future. 
+The best way to predict the future is to look at the past!
+Learning a new tool is an investment, in addition to the time spent integrating it into your workflows.
+If a tool improves quickly over time, a simple version update can speed up your analysis.
+
+When choosing an analytical engine, you also want to consider the situation where your data sizes could grow substantially.
+If the system is improving rapidly, then the system can scale as your data scales, with lower chances for a migration in the future.
+Plus, if you haven't experimented with a tool in a while, you can see how much faster it has become since you last checked!
+
+#### What is it especially good at?
+
+The choice of benchmark is an indicator of what types of workloads a tool is useful for. 
+The higher the variety of analyses in the benchmark, the more broadly useful the tool can be.
+
+#### What scale of data can it handle?
+
+Many benchmarks are deliberately smaller than typical workloads. 
+This allows the benchmark to complete in a reasonable amount of time when run with many configurations. 
+However, an important question to answer when selecting a system is whether the size of your data can be handled within the size of your compute resources.
+
+### Limitations of Benchmarking over Time
+
+There are some limitations when looking at the performance of a system over time.
+If a feature is brand new, there is no prior performance to compare to!
+As a result, this post focuses on fundamental workloads rather than DuckDB's ever-increasing set of integrations with different lakehouse data formats, cloud services, and more.
+
+The code used to run the benchmark also avoids many of DuckDB's [Friendlier SQL](https://duckdb.org/docs/guides/sql_features/friendly_sql) additions, as those have also been added more recently. 
+(When writing these queries, it felt like going back in time!)
+
+
+## Benchmark Design
+
+### Summary
+
+* H2O.ai, plus import/export and window function tests
+* Python instead of R
+* 5GB scale for everything, plus 50GB scale for group bys and joins
+* Median of 3 runs
+* Using a MacBook Pro M1 with 16GB RAM
+* DuckDB Versions 0.2.7 through 1.0.0
+    * Nearly 3 years, from 2021-06-14 to 2024-06-03
+* Default settings
+
+
+### H2O.ai as the Foundation
+
+This post measures DuckDB's performance over time on the H2O.ai benchmark for both joins and group by queries.
+Please see our previous [blog](https://duckdb.org/2023/04/14/h2oai.html) [posts](https://duckdb.org/2023/11/03/db-benchmark-update.html) for details on why we believe the H2O.ai benchmark is a good approach!
```

Comment:
H20.ai has been updated recently, I will also put the stable website somewhere.

Reply:
Oh, thank you!

Reply:
I added the link you sent - thanks!